Rewrite Scripting section of the User Manual (#644)
playcanvas/developer-site: docs/user-manual/scripting/migration-guide.md

And here is the equivalent script in the current format:

import Tabs from '@theme/Tabs';
import TabItem from '@theme/TabItem';

<Tabs defaultValue="classic" groupId='script-code'>
<TabItem  value="esm" label="ESM">

```javascript

> import { ScriptType } from 'playcanvas';
> 
> export class MyScript extends ScriptType {

Reviewer: ```suggestion
import { Script } from 'playcanvas';

export class MyScript extends Script {
```

To migrate these to the current script format, you would register event handlers in the script's `initialize` function:

<Tabs defaultValue="classic" groupId='script-code'>
<TabItem  value="esm" label="ESM">

```javascript

> import { ScriptType } from 'playcanvas';
> 
> export class Movement extends ScriptType {

Reviewer: ```suggestion
import { Script } from 'playcanvas';

export class Movement extends Script {
```

    initialize() {

    }

    update(dt) {
        this.entity.setPosition(Math.sin(Date.now() / 1000), 0.5, 0);
    }
}
```

</TabItem>
<TabItem value="classic" label="Classic">

```javascript

> export class MyScript extends ScriptType {
>     initialize() {
>         this.on("enable", function () {

Reviewer: ```suggestion
export class MyScript extends Script {
    initialize() {
        this.on("enable", () => {
```

        });

>         this.on("disable", function () {
> 
>         });
> 
>         this.on("destroy", function () {

Reviewer: ```suggestion
        this.on("disable", () => {

        });

        this.on("destroy", () => {
```
